typhos/positioner.py

```diff
@@ -654,18 +732,260 @@ def update_alarm_text(self, alarm_level):
+        except KeyError:
+            text = self.alarm_text[alarms.INVALID]
         self.ui.alarm_label.setText(text)
 
+    @property
+    def all_linked_attributes(self) -> list[str]:
+        """All linked attribute names."""
+        return [
+            attr
+            for attr in (
+                self.acceleration_attribute,
+                self.error_message_attribute,
+                self.high_limit_switch_attribute,
+                self.high_limit_travel_attribute,
+                self.low_limit_switch_attribute,
+                self.low_limit_travel_attribute,
+                self.moving_attribute,
+                self.readback_attribute,
+                self.setpoint_attribute,
+                self.velocity_attribute,
+            )
+            if attr
+        ]
+
+    @property
+    def all_linked_signals(self) -> list[ophyd.Signal]:
+        """All linked signal names."""
+        signals = [
+            getattr(self.device, attr, None)
+            for attr in self.all_linked_attributes
+        ]
+        return [sig for sig in signals if sig is not None]
+
+    def show_ui_type_hints(self):
+        """Show type hints of widgets included in the UI file."""
+        cls_attrs = set()
+        obj_attrs = set(dir(self.ui))
+        annotated = set(self.ui.__annotations__)
+        for cls in type(self.ui).mro():
+            cls_attrs |= set(dir(cls))
+        likely_from_ui = obj_attrs - cls_attrs - annotated
+        for attr in sorted(likely_from_ui):
+            try:
+                obj = getattr(self, attr, None)
+            except Exception:
+                ...
+            else:
+                if obj is not None:
+                    print(f"{attr}: {obj.__class__.__module__}.{obj.__class__.__name__}")
+
+
+class _TyphosPositionerRowUI(_TyphosPositionerUI):
+    """Annotations helper for positioner_row.ui; not to be instantiated."""
+
+    notes_edit: TyphosNotesEdit
+    status_container_widget: QtWidgets.QFrame
+    extended_signal_panel: Optional[TyphosSignalPanel]
+    error_prefix: QtWidgets.QLabel
+
+
+class TyphosPositionerRowWidget(TyphosPositionerWidget):
+    ui: _TyphosPositionerRowUI
+    ui_template = os.path.join(utils.ui_dir, "widgets", "positioner_row.ui")
+
+    alarm_text = {
+        AlarmLevel.NO_ALARM: 'ok',
+        AlarmLevel.MINOR: 'minor',
+        AlarmLevel.MAJOR: 'major',
+        AlarmLevel.DISCONNECTED: 'conn',
+        AlarmLevel.INVALID: 'inv',
+    }
+
+    def __init__(self, *args, **kwargs):
+        self._error_message = ""
+        self._status_text = ""
+        self._alarm_level = AlarmLevel.DISCONNECTED
+
+        super().__init__(*args, **kwargs)
+
+        for idx in range(self.layout().count()):
+            item = self.layout().itemAt(idx)
+            if item is self.ui.status_text_layout:
+                self.layout().takeAt(idx)
+                break
+
+        # TODO move these out
+        self._omit_names = [
```

Comment:
The idea is that they're already part of the positioner row, so there's no need to show them again in the signal listing in the "expanded details" (i.e., after clicking `>`)

Reply:
So there's no more `# TODO` then?

Reply:
Oh, I didn't notice you were commenting on that part. Good question - the `TODO` was pointing at the fact that these should probably be part of a Qt designable property instead of hard-coded in `typhos.positioner`

It should be rephrased or made into a more verbose issue I suppose